### PR TITLE
User functions

### LINKED ISSUE #44: Favourites
# User-Story

As a user

I want to save favourites

so that i can look at them later

# Notes, Links, Material

# ToDos

- [x] #43 
- [x] save to database function
- [x] delete from database function

### PR DESCRIPTION
**Pull request by kschmitz123**
- If user = creator , display delete button
- added basic login function

todo :
- [x] save user favourites
- [x]  add delete project function
- [x] change heart color when clicked
- [x] delete from favorites

close #44

### CODE REVIEW

File: client/src/components/Header.js [hunk 1]
```diff
@@ -21,25 +24,21 @@ const StyledHeader = styled.header`
...
 const Header = ({ title }) => {
   const location = useLocation();
+  const history = useHistory();
+  const { logout } = useUserState();
+
+  const handleClick = () => {
+    logout();
+    history.push("/");
+  };
   useEffect(() => {}, [location]);
```
Comment (AlexisRoe) [inline client/src/components/Header.js]: short question, what is the useEffect for, because its empty
```suggestion;
```

**Review by AlexisRoe — COMMENTED** [review]
good job👍

File: client/src/components/Header.js (hunk 1 above)
Comment (kschmitz123) [reply client/src/components/Header.js]: good point. i removed it because it had no effect 👍

**Review by phlgr — APPROVED** [review]
Nice, clean code. LGTM 🚀

**Review by arassaria — COMMENTED** [review]
LGTM 🚀

**Review by mariegross — APPROVED** [review]
lgtm – great progress 🥳🤩

File: client/src/pages/DetailsPage.js [hunk 2]
```diff
@@ -26,11 +28,31 @@ const Title = styled.h3`
 export const DetailsPage = () => {
   const { projectId } = useParams();
   const { user } = useUserState();
+  const [favoriteURL, setFavoriteURL] = useState("");
+  const [favoriteID, setFavoriteID] = useState("");
+  const [favoriteTitle, setFavoriteTitle] = useState("");
```
Comment (marrep) [inline client/src/pages/DetailsPage.js]: I am not familiar with your project yet, but wouldn't it be easier to compile those states into one? (object)

Comment (kschmitz123) [reply client/src/pages/DetailsPage.js]: it would probably be easier, i'll have a look at that again

**Review by phlgr — APPROVED** [review]
LGTM 🚀

**Pull request merged**